great_expectations/data_context/data_context.py

```diff
@@ -636,12 +557,14 @@ def add_datasource(self, name, type_, **kwargs):
         if not "datasources" in self._project_config:
             self._project_config["datasources"] = {}
         self._project_config["datasources"][name] = datasource.get_config()
-        self._save_project_config()
 
-        return datasource
+        #!!! This return value isn't used anywhere in the live codebase, and only once in tests.
```

Comment:
At the time, I thought that including a return value would make it much harder to subclass this method. Now I think I see a way to do it.

That said, AFACIT, we're not using that functionality anywhere, and it's not parallel with the way we implemented `add_store`. Why keep it?

As long as the subclassing issue really is gone, I don't feel strongly about this either way,

Reply:
I'm flagging this as an example of scope creep that makes the PR hard to review -- the return value of add_datasource() shouldn't be a concern of a PR that adds a store feature, and it makes any review need to be done more closely when there are subtle changes introduced.

The main reason to keep this is that when we've just dynamically loaded a class and instantiated it internally, we don't want a user to potentially have to do that again in order to use the object that was just created.

With respect to reasons--I'm not sure what the subclassing issue was that seemed problematic, but it doesn't make sense to me to say "it's not parallel with the way we implemented `add_store`" because this is the PR that implements that...this PR is breaking parallelism.